app/src/main/java/org/wikipedia/analytics/eventplatform/UserContributionEvent.kt

```diff
@@ -0,0 +1,108 @@
+package org.wikipedia.analytics.eventplatform
+
+import com.google.gson.annotations.SerializedName
+import org.wikipedia.WikipediaApp
+
+class UserContributionEvent private constructor() : Event(SCHEMA_NAME, STREAM_NAME) {
+    @SerializedName("action") private var action: String = ""
+
+    fun logOpen() {
+        action = "open_hist"
+        submitEvent()
+    }
+
+    fun logFilterDescriptions() {
+        action = "filt_desc"
+        submitEvent()
+    }
+
+    fun logFilterCaptions() {
+        action = "filt_caption"
+        submitEvent()
+    }
+
+    fun logFilterTags() {
+        action = "filt_tag"
+        submitEvent()
+    }
+
+    fun logFilterAll() {
+        action = "filt_all"
+        submitEvent()
+    }
+
+    fun logViewDescription() {
+        action = "desc_view"
+        submitEvent()
+    }
+
+    fun logViewCaption() {
+        action = "caption_view"
+        submitEvent()
+    }
+
+    fun logViewTag() {
+        action = "tag_view"
+        submitEvent()
+    }
+
+    fun logViewMisc() {
+        action = "misc_view"
+        submitEvent()
+    }
+
+    fun logNavigateDescription() {
+        action = "desc_view2"
+        submitEvent()
+    }
+
+    fun logNavigateCaption() {
+        action = "caption_view2"
+        submitEvent()
+    }
+
+    fun logNavigateTag() {
+        action = "tag_view2"
+        submitEvent()
+    }
+
+    fun logNavigateMisc() {
+        action = "misc_view2"
+        submitEvent()
+    }
+
+    fun logPaused() {
+        action = "paused"
+        submitEvent()
+    }
+
+    fun logDisabled() {
+        action = "disabled"
+        submitEvent()
+    }
+
+    fun logIpBlock() {
+        action = "ip_block"
+        submitEvent()
+    }
+
+    private fun submitEvent() {
+        WikipediaApp.getInstance().eventPlatformClient.submit(this)
+    }
+
+    companion object {
+        private var INSTANCE: UserContributionEvent? = null
```

Comment:
I understand you were modeling this class on the old `UserContributionFunnel`, but this class must _absolutely not_ be a singleton.